source/agora/node/Ledger.d

```diff
@@ -142,14 +155,21 @@ public class Ledger
     ***************************************************************************/
 
     public const(Block)[] getBlocksFrom (ulong block_height, size_t max_blocks)
-        @safe nothrow @nogc
+        @safe
     {
         assert(max_blocks > 0);
 
-        if (block_height > this.ledger.length)
-            return null;
+        Block[] res;
+        Block block;
+        foreach (height;  block_height .. block_height + max_blocks)
+        {
+            if (this.ledger.readBlock(block, height))
+                res ~= block;
+            else
+                break;
+        }
 
-        return this.ledger[block_height .. min(block_height + max_blocks, $)];
+        return cast(const(Block)[])res;
```

Comment:
You could just change `Block[] res;` into `const(Block)[]` res, and return it here without casting.

Reply:
modified